torchsparse/nn/functional/conv.py

```diff
@@ -15,6 +16,7 @@
 
 class SpConvolution(Function):
     @staticmethod
+    @custom_fwd(cast_inputs=torch.half)
```

Comment:
I'm wondering if we need to specify `cast_inputs`. I'm not that familiar with the default behavior of `custom_fwd`.

Reply:
I think it was necessary, if you don't specify `cast_inputs` it will remain in the current autocast state, which will be float32 unless some function before it gets autocasted to float16. But if autocast isn't enabled, the cast_inputs becomes a no-op, so it won't affect float32 training (the way they handle custom autograd functions with autocast is pretty weird, it took a bit of experimenting to get it to work)

Reply:
Thanks for the clarification!